bundles/framework/announcements/tool/AnnouncementsTool.js

```diff
@@ -0,0 +1,367 @@
+                    me._sendAnnouncementsChangedEvent(me.selectedAnnouncements);
+                }
+                jQuery('div.basic_publisher').find('input[name=publisher-announcements]').val(me.annTitles.toString()).attr('announcement-name', me.annTitles.toString());
+            });
+
+            popup.show(title, content, [closeButton]);
+            this._announcementsPopup = popup;
+            me.isAnnouncementsDialogOpen = true;
+        },
+
+        /**
+         * Should preselect announcement.
+         * @method @private shouldPreselectAnnouncement
+         * @param  {Integer} id announcement id
+         * @return {Boolean} true if announcement must be preselect, other false
+         */
+        shouldPreselectAnnouncement: function (id) {
+            const toolPluginMapfullConf = this._getToolPluginMapfullConf();
+            if (toolPluginMapfullConf) {
+                var isPluginConfig = !!((toolPluginMapfullConf && toolPluginMapfullConf.config &&
+                    toolPluginMapfullConf.config.announcements));
+
+                if (isPluginConfig) {
+                    for (var i = 0; i < toolPluginMapfullConf.config.announcements.length; i++) {
+                        if (toolPluginMapfullConf.config.announcements[i].id == '' + id) {
+                            return true;
+                        }
+                    }
+                } else {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        },
+        /**
+         * @private @method _getToolPluginMapfullConf
+         * Get map view cofiguration (from mapfull) for this tool
+         * @return {Object / null} config or null if not found
+         */
+        _getToolPluginMapfullConf: function () {
+            var me = this;
+            var isConfig = !!((me.data && me.data.configuration));
+            var isPlugins = !!((isConfig && me.data.configuration.mapfull &&
+            me.data.configuration.mapfull.conf && me.data.configuration.mapfull.conf.plugins));
+            var toolPlugin = null;
+            if (isPlugins) {
+                var plugins = me.data.configuration.mapfull.conf.plugins;
+                for (var i = 0; i < plugins.length; i++) {
+                    var plugin = plugins[i];
+                    if (plugin.id === me.getTool().id) {
+                        toolPlugin = plugin;
+                        break;
+                    }
+                }
+            }
+            return toolPlugin;
+        },
+
+        fetchAnnouncements: function () {
+            jQuery.ajax({
+                type: 'GET',
+                dataType: 'json',
+                url: Oskari.urls.getRoute('Announcements'),
+                success: (pResp) => {
+                    this.announcements = pResp.data;
+                },
+                error: function (jqXHR, textStatus) {
+                }
+            });
+        },
```

Comment:
Aren't announcements already loaded to the frontend if the appsetup has announcements bundle included? Maybe refactor to a service and get the data from there?